### ISSUE TITLE
Coding style

### ISSUE THREAD

**Issue by Janluke0** [frittatelle/oxycloud-frontend#6]
We need to estabilsh conventions like:

- use came case for functions
- snake case for variables
- max 80 char length for lines
- not more than a newline in a row

and configure a plugin of vscode to verify/format the code according to them

**Comment by frittatelle** [comment]
- Please specify components as arrow functions

*Component template*:
```
const Component = ({props}) => {
    //Component functions
    const foo = () => {
        return bar; 
    }
    //Rendering part
    return(
        //Object to render
        <div>
            <FooComponent />
            <BarComponent />
        </div>
    );
}

export default Component;
```
------
- Please always specify the state as [state,setState] and give a default value:

*State template*:
```
const [state,setState] = useState('default_value');
```
---
- Try to minimize the number of props passed to child components
- Access props by using their names and not by using props.name

**Comment by Janluke0** [comment]
I tried to apply the arrows style in 7093f07.

It works (actually the next commit) but it wasn't easy to apply:
The class approch offers method called in differents phase of the life cycle of the component like the constructor, *ComponentDidMount*, *ComponentDidUpdate* (which even receive as parameters what was changed in the state).

I, very personally, find the classes more natural.

Now it shows some warings by compiler (is in the comments) and by the browser (some callback requires toomuch time). It's probably my bad because i simply can't use it in the right way but i think that rattle off the problem could be usefull for everybody

**Comment by frittatelle** [comment]
One of the benefits of using arrow functions is that it will make simpler the use of *react-query* that manages easily the component life cycle. 
**useQuery** from *react-query* actually returns three objects (data, status, error) that can be used to render the ui accordingly.
*Status* can be used to achieve the same objective of *ComponentDidMount* etc.:

*react-query state and status management example*

```Javascript
const Planets = () => {
    
    const {data, status, error} = useQuery('planets', fetchPlanets);

    return(
        <>
            {status === 'loading' && (
                <h1>loading data ...</h1>
            )}
            {status === 'error' && (
                <h1>error fetching data: {error}</h1>
            )}
            {status === 'success' && (
                <>
                    <h1>Data successfully retrieved</h1>
                    <div>{data}</div>
                </>
            )}
        </>
    );
}

export default Planets;
```

**Comment by frittatelle** [comment]
# Utility/Help functions and components

I looked up some different resources to find React best practices to deal with this kind of functions and components. 
There are two different approaches:
- Creating a new module that contains all the utility functions and that can be imported and used at will.
- Creating a /utils directory on the same level as the /components directory which contains functions that can be used at will.

What do you think it's the best solution for us?

**Comment by Janluke0** [comment]
I'd prefer the second one. 

Actually i did it (😅), for the s3 api wrapper, but calling it *libs*.  
Utils make more sense for the functions for date and files size formatting.

I can rename it and we can define these small functions inside `index.js` 
so we can import them like `import {formatter} from '../utils'`  and leaving more complex things inside dedicated files
to import the like `import Storage from '../utils/storage-api'.js`
This approch is more flexible because if we start we a little things that grow unexpectedly, we can just move it in another file and import and export inside `index.js` so component will not requires an update.

Anyway i don't like too much the name of that little *lib*.
Maybe somethings like `import {OxyStorage} from '../utils/api'` could be cooler, what do you think?

TL;dr;  
the second one
PS i build the s3 api wrapper, and merged your and @nucciocrea9  branch inside dev-file_explorer (4fd3c7f) i did some little refactor and edit to make it work and pass the CI build and tests.

**Comment by frittatelle** [comment]
Nice! I'll check it and the api out tomorrow.

**Comment by nucciocrea9** [comment]
I think is better to use utils for the functions which define date and the size of the file
